plugins/discourse-local-dates/assets/javascripts/lib/rich-editor-extension.js

```diff
@@ -0,0 +1,188 @@
+/** @type {RichEditorExtension} */
+const extension = {
+  // TODO(renato): the rendered date needs to be localized to better match the cooked content
```

Comment:
I think until we do this, it might be prudent to add the timezone as well to the output. So instead of:

```
2021-01-01 12:00:00
```

We have

```
2021-01-01 12:00:00 (America/New York)
```

_If_ the timezone is specified in the original markdown.

I see we already have a way to pass in extra things like session in plugins:

https://github.com/discourse/discourse/blob/39eb5df6d9898ddbbd354af7b35d1a3fa92fc584/app/assets/javascripts/discourse/app/static/prosemirror/components/prosemirror-editor.gjs#L84-L89

Do we already have something to do this for extensions? If not we will need it for `currentUser` here, and I see in other extensions you have the need for `siteSettings` too.

Reply:
Yes, this would be the right place to expose this kind of context 👍 

Let's move forward as-is but re-visit all TODOs right after